### ISSUE TITLE
Refactor ThreadsMaintenanceController

### ISSUE THREAD

**Issue by shamrice** [shamrice/discapp#473]
The postDiscEditView method specifically is a horror. It is currently around 550 lines long and handles all POST calls possible on the thread admin section. 
If method still takes in all actions, those actions should be broken into their own manageable methods. This was already done with the delete and delete message and replies buttons.

**Comment by shamrice** [comment]
Adding this to release 4.2.0. There was more work done in this section in release 4.1.0 and doing so was a nightmare thanks to this controller being such a mess. 4.2.0 has additional changes to the thread admin section and is a perfect time to finally get this done.

**Comment by shamrice** [comment]
Moving to 4.3.0 as 4.2.0 release has a bunch of other tickets in it now.

**Comment by shamrice** [comment]
Going to call this one done for now. Large changes have been made to to the POST actions for thread maintenance. Will wait on any additional changes to the GET actions (if needed).
